src/devices/Lp55231/Lp55231.cs

```diff
@@ -0,0 +1,168 @@
+        /// </summary>
+        /// <param name="settings">The I2C settings of the Lp55231.</param>
+        /// <remarks>
+        /// The default i2c address is 0x60, but the HAT can be configured in hardware to any address from 0x60 to 0x7f.
+        /// The PWM hardware used by this HAT is a PCA9685. It has a total possible frequency range of 24 to 1526 Hz.
+        /// Setting the frequency above or below this range will cause PWM hardware to be set at its maximum or minimum setting.
+        /// </remarks>
+        public Lp55231(I2cConnectionSettings? settings = default)
+        {
+            settings ??= new I2cConnectionSettings(1, I2cAddressBase);
+
+            _device = I2cDevice.Create(settings);
+
+            _rgbLeds = new[]
+            {
+                new RgbLed(0, SetIntensity),
+                new RgbLed(1, SetIntensity),
+                new RgbLed(2, SetIntensity)
+            };
+        }
+
+        private byte ReadRegister(Register register)
+        {
+            return ReadRegister((byte)register);
+        }
+
+        private byte ReadRegister(byte register)
+        {
+            _device.WriteByte((byte)register);
+            return _device.ReadByte();
+        }
+
+        private void WriteRegister(Register register, byte data)
+        {
+            WriteRegister((byte)register, data);
+        }
+
+        private void WriteRegister(byte register, byte data)
+        {
+            Span<byte> bytes = stackalloc byte[2];
+            bytes[0] = (byte)register;
+            bytes[1] = data;
+            _device.Write(bytes);
+        }
+
+        private void SetIntensity(byte index, byte value)
+        {
+            byte register = (byte)(Register.REG_D1_PWM + index);
+            WriteRegister(register, value);
+        }
+
+        /// <inheritdoc/>
+        public void Dispose()
+        {
+            _device.Dispose();
+        }
+
+        /// <summary>
+        /// Resets the Lp55231
+        /// </summary>
+        /// <remarks>
+        /// You should delay after calling this method
+        /// </remarks>
+        public void Reset()
+        {
+            try
+            {
+                WriteRegister(Register.REG_RESET, 0xFF);
+            }
+            catch (IOException)
+            {
+                // Resetting will prevent the expected ack resulting
+                // in an IOException. It's fine to swallow this.
+            }
+        }
+
+        /// <summary>
+        /// Gets/sets whether the Lp55231 is enabled.
+        /// </summary>
+        /// <remarks>
+        /// Setting this value will stop eny programs currently running.
+        /// </remarks>
+        public bool Enabled
+        {
+            get
+            {
+                var register = ReadRegister(Register.REG_CNTRL1);
+                return (register & (byte)Control1RegisterFlags.Enabled) > 0;
+            }
+            set
```

Comment:
you're missing an enter here. I'm surprised the build didn't raised it :-)

Reply:
Where is the missing enter? Layout looks correct to me??

Reply:
between } and set. There should be a blank new line after a }

Reply:
Done